File: Samples/VB.Net/VBDotNet/ISecurityDlg.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports SalesMatePlusLib

Namespace CSharpSDKTest
	Partial Public Class ISecurityDlg
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Accesscstmrbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles Accesscstmrbtn.Click
			Dim m_Security As ISecurity
			m_Security = New SecurityClass()
			Infolist.Items.Clear()
			InfoLbl.Text = ""
			Securitylist.Items.Clear()
			Dim lAcesscstmer As Long
			lAcesscstmer=m_Security.AccessCustomerMenu
			If lAcesscstmer = 1 Then
				Infolist.Items.Add("Have rights to access customer menu")
			ElseIf lAcesscstmer = 0 Then
				Infolist.Items.Add("Access to customer menu is denied")
			End If
		End Sub

		Private Sub Acesssalesbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles Acesssalesbtn.Click
			Dim m_Security As ISecurity
			m_Security = New SecurityClass()
			Infolist.Items.Clear()
			InfoLbl.Text = ""
			Securitylist.Items.Clear()
			Dim lAcessSals As Long
			lAcessSals = m_Security.AccessSalesMenu
			If lAcessSals = 1 Then
				Infolist.Items.Add("Have rights to access sales menu")
			ElseIf lAcessSals = 0 Then
				Infolist.Items.Add("Access to sales menu is denied")
			End If

		End Sub

		Private Sub AccessSetupbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles AccessSetupbtn.Click
			Dim m_Security As ISecurity
			m_Security = New SecurityClass()
			Infolist.Items.Clear()
			InfoLbl.Text = ""
			Securitylist.Items.Clear()
			Dim lAcessSetup As Long
			lAcessSetup = m_Security.AccessSetup
			If lAcessSetup = 1 Then
				Infolist.Items.Add("Have rights to access setup")
			ElseIf lAcessSetup = 0 Then
				Infolist.Items.Add("Access to setup is denied")
			End If
		End Sub

		Private Sub Accesstransctnbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles Accesstransctnbtn.Click
			Dim m_Security As ISecurity
			m_Security = New SecurityClass()
			Infolist.Items.Clear()
			InfoLbl.Text = ""
			Securitylist.Items.Clear()
			Dim lAcessTransaction As Long
			lAcessTransaction = m_Security.AccessTransactionMenu
			If lAcessTransaction = 1 Then
				Infolist.Items.Add("Have rights to access transaction menu")
			ElseIf lAcessTransaction = 0 Then
				Infolist.Items.Add("Access to transaction menu is denied")
			End If
		End Sub

		Private Sub CrntusrLoginbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles CrntusrLoginbtn.Click
			Dim m_Security As ISecurity
			m_Security = New SecurityClass()
			Infolist.Items.Clear()
			InfoLbl.Text = ""
			Securitylist.Items.Clear()
			Dim strCurentUsrLogin As String
			strCurentUsrLogin = m_Security.CurrentLoginUser
			If strCurentUsrLogin = "" Then
				Infolist.Items.Add("No current user")
			Else
				Infolist.Items.Add(strCurentUsrLogin)
			End If
		End Sub

		Private Sub Crntusrbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles Crntusrbtn.Click
			Dim m_Security As ISecurity
			m_Security = New SecurityClass()
			Infolist.Items.Clear()
			InfoLbl.Text = ""
			Securitylist.Items.Clear()
			Dim strCrntUsrRights As String
			strCrntUsrRights = m_Security.CurrentUserRights
			If strCrntUsrRights = "" Then
				Infolist.Items.Add("No current user rights")
			Else
				Infolist.Items.Add(strCrntUsrRights)
			End If
		End Sub

		Private Sub DatabaseRytbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles DatabaseRytbtn.Click
			Dim m_Security As ISecurity
			m_Security = New SecurityClass()
			Infolist.Items.Clear()
			InfoLbl.Text = ""
			Securitylist.Items.Clear()
			Dim lDatabseRite As Long
			lDatabseRite = m_Security.DatabaseRight
			If lDatabseRite = 1 Then
				Infolist.Items.Add("Have database right")
			ElseIf lDatabseRite = 0 Then
				Infolist.Items.Add("Database right is denied")
			End If
		End Sub

		Private Sub PrntRptsbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles PrntRptsbtn.Click
			Dim m_Security As ISecurity
			m_Security = New SecurityClass()
			Infolist.Items.Clear()
			InfoLbl.Text = ""
			Securitylist.Items.Clear()
			Dim lPrntrpts As Long
			lPrntrpts = m_Security.PrintReports
			If lPrntrpts = 1 Then
				Infolist.Items.Add("Have print rights")
			ElseIf lPrntrpts = 0 Then
				Infolist.Items.Add("Doesn't have print rights")
			End If
		End Sub

		Private Sub ShpbalBtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles ShpbalBtn.Click
			Dim m_Security As ISecurity
			m_Security = New SecurityClass()
			Infolist.Items.Clear()
			InfoLbl.Text = ""
			Securitylist.Items.Clear()
			Dim dShpBal As Double
			dShpBal = m_Security.UpdateShopBalance
			Infolist.Items.Add(dShpBal)
		End Sub

		Private Sub VwRptBtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles VwRptBtn.Click
			Dim m_Security As ISecurity
			m_Security = New SecurityClass()
			Infolist.Items.Clear()
			InfoLbl.Text = ""
			Securitylist.Items.Clear()
			Dim lviewRpts As Long
			lviewRpts = m_Security.ViewReports
			If lviewRpts = 1 Then
				Infolist.Items.Add("Have rights to view reports")
			ElseIf lviewRpts = 0 Then
				Infolist.Items.Add("No rights view reports")
			End If
		End Sub

		Private Sub UsrRytBtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles UsrRytBtn.Click
			Dim m_Security As ISecurity
			m_Security = New SecurityClass()
			InfoLbl.Text = ""
			Infolist.Items.Clear()
			Securitylist.Items.Clear()
			Dim strXML As String
			Dim strModule As String
			Dim strLeaf As String
			Dim lUserRyt As Long
			strXML=XMTTxt.Text
			strModule=ModuleTxt.Text
			strLeaf=LeafsecurityTxt.Text
			If (strXML = "") AndAlso (strModule = "") AndAlso (strLeaf = "") Then
				MessageBox.Show("Please the XML Module name,module section and leaf security node name")
			ElseIf strXML = "" Then
				MessageBox.Show("Please enter the XML module name")
			ElseIf strModule = "" Then
				MessageBox.Show("Please enter the module section")
			ElseIf strLeaf = "" Then
				MessageBox.Show("Please enter the leaf security node name")
			Else
				lUserRyt= m_Security.UserRightInfo(strXML, strModule, strLeaf)
				If lUserRyt = 1 Then
					InfoLbl.Text = "Have User Right Info"
				ElseIf lUserRyt = 0 Then
					InfoLbl.Text = "Don't have user right Info"
				End If
			End If

		End Sub

		Private Sub UserBtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles UserBtn.Click
			Dim m_Security As ISecurity
			m_Security = New SecurityClass()
			InfoLbl.Text = ""
			Securitylist.Items.Clear()
			Infolist.Items.Clear()
			Dim strUserName As String
			Dim strUserSecurity As String=""
			strUserName = UsrNmeTxt.Text
			If strUserName = "" Then
				MessageBox.Show("Please enter the user name")
			Else
				m_Security.GetUserRightsInfo(strUserName, strUserSecurity)
				Securitylist.Items.Add(strUserSecurity)
			End If
		End Sub


	End Class
End Namespace

File: Samples/C#.Net/CSharpSDKTest/ISecurityDlg.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using SalesMatePlusLib;

namespace CSharpSDKTest
{
    public partial class ISecurityDlg : Form
    {
        public ISecurityDlg()
        {
            InitializeComponent();
        }

        private void Accesscstmrbtn_Click(object sender, EventArgs e)
        {
            ISecurity m_Security;
            m_Security = new SecurityClass();
            Infolist.Items.Clear();
            InfoLbl.Text = "";
            Securitylist.Items.Clear();
            long lAcesscstmer;
            lAcesscstmer=m_Security.AccessCustomerMenu;
            if (lAcesscstmer == 1)
            {
                Infolist.Items.Add("Have rights to access customer menu");
            }
            else if (lAcesscstmer == 0)
            {
                Infolist.Items.Add("Access to customer menu is denied");
            }
        }

        private void Acesssalesbtn_Click(object sender, EventArgs e)
        {
            ISecurity m_Security;
            m_Security = new SecurityClass();
            Infolist.Items.Clear();
            InfoLbl.Text = "";
            Securitylist.Items.Clear();
            long lAcessSals;
            lAcessSals = m_Security.AccessSalesMenu;
            if (lAcessSals == 1)
            {
                Infolist.Items.Add("Have rights to access sales menu");
            }
            else if (lAcessSals == 0)
            {
                Infolist.Items.Add("Access to sales menu is denied");
            }

        }

        private void AccessSetupbtn_Click(object sender, EventArgs e)
        {
            ISecurity m_Security;
            m_Security = new SecurityClass();
            Infolist.Items.Clear();
            InfoLbl.Text = "";
            Securitylist.Items.Clear();
            long lAcessSetup;
            lAcessSetup = m_Security.AccessSetup;
            if (lAcessSetup == 1)
            {
                Infolist.Items.Add("Have rights to access setup");
            }
            else if (lAcessSetup == 0)
            {
                Infolist.Items.Add("Access to setup is denied");
            }
        }

        private void Accesstransctnbtn_Click(object sender, EventArgs e)
        {
            ISecurity m_Security;
            m_Security = new SecurityClass();
            Infolist.Items.Clear();
            InfoLbl.Text = "";
            Securitylist.Items.Clear();
            long lAcessTransaction;
            lAcessTransaction = m_Security.AccessTransactionMenu;
            if (lAcessTransaction == 1)
            {
                Infolist.Items.Add("Have rights to access transaction menu");
            }
            else if (lAcessTransaction == 0)
            {
                Infolist.Items.Add("Access to transaction menu is denied");
            }
        }

        private void CrntusrLoginbtn_Click(object sender, EventArgs e)
        {
            ISecurity m_Security;
            m_Security = new SecurityClass();
            Infolist.Items.Clear();
            InfoLbl.Text = "";
            Securitylist.Items.Clear();
            string strCurentUsrLogin;
            strCurentUsrLogin = m_Security.CurrentLoginUser;
            if (strCurentUsrLogin == "")
            {
                Infolist.Items.Add("No current user");
            }
            else
                Infolist.Items.Add(strCurentUsrLogin);
        }

        private void Crntusrbtn_Click(object sender, EventArgs e)
        {
            ISecurity m_Security;
            m_Security = new SecurityClass();
            Infolist.Items.Clear();
            InfoLbl.Text = "";
            Securitylist.Items.Clear();
            string strCrntUsrRights;
            strCrntUsrRights = m_Security.CurrentUserRights;
            if (strCrntUsrRights == "")
            {
                Infolist.Items.Add("No current user rights");
            }
            else
                Infolist.Items.Add(strCrntUsrRights);
        }

        private void DatabaseRytbtn_Click(object sender, EventArgs e)
        {
            ISecurity m_Security;
            m_Security = new SecurityClass();
            Infolist.Items.Clear();
            InfoLbl.Text = "";
            Securitylist.Items.Clear();
            long lDatabseRite;
            lDatabseRite = m_Security.DatabaseRight;
            if (lDatabseRite == 1)
            {
                Infolist.Items.Add("Have database right");
            }
            else if (lDatabseRite == 0)
            {
                Infolist.Items.Add("Database right is denied");
            }
        }

        private void PrntRptsbtn_Click(object sender, EventArgs e)
        {
            ISecurity m_Security;
            m_Security = new SecurityClass();
            Infolist.Items.Clear();
            InfoLbl.Text = "";
            Securitylist.Items.Clear();
            long lPrntrpts;
            lPrntrpts = m_Security.PrintReports;
            if (lPrntrpts == 1)
            {
                Infolist.Items.Add("Have print rights");
            }
            else if (lPrntrpts == 0)
            {
                Infolist.Items.Add("Doesn't have print rights");
            }
        }

        private void ShpbalBtn_Click(object sender, EventArgs e)
        {
            ISecurity m_Security;
            m_Security = new SecurityClass();
            Infolist.Items.Clear();
            InfoLbl.Text = "";
            Securitylist.Items.Clear();
            double dShpBal;
            dShpBal = m_Security.UpdateShopBalance;
            Infolist.Items.Add(dShpBal);
        }

        private void VwRptBtn_Click(object sender, EventArgs e)
        {
            ISecurity m_Security;
            m_Security = new SecurityClass();
            Infolist.Items.Clear();
            InfoLbl.Text = "";
            Securitylist.Items.Clear();
            long lviewRpts;
            lviewRpts = m_Security.ViewReports;
            if (lviewRpts == 1)
            {
                Infolist.Items.Add("Have rights to view reports");
            }
            else if (lviewRpts == 0)
            {
                Infolist.Items.Add("No rights view reports");
            }
        }

        private void UsrRytBtn_Click(object sender, EventArgs e)
        {
            ISecurity m_Security;
            m_Security = new SecurityClass();
            InfoLbl.Text = "";
            Infolist.Items.Clear();
            Securitylist.Items.Clear();
            string strXML;
            string strModule;
            string strLeaf;
            long lUserRyt;
            strXML=XMTTxt.Text;
            strModule=ModuleTxt.Text;
            strLeaf=LeafsecurityTxt.Text;
            if ((strXML == "") && (strModule == "") && (strLeaf == ""))
            {
                MessageBox.Show("Please the XML Module name,module section and leaf security node name");
            }
            else if (strXML == "")
            {
                MessageBox.Show("Please enter the XML module name");
            }
            else if (strModule == "")
            {
                MessageBox.Show("Please enter the module section");
            }
            else if (strLeaf == "")
            {
                MessageBox.Show("Please enter the leaf security node name");
            }
            else
            {
                lUserRyt= m_Security.get_UserRightInfo(strXML, strModule, strLeaf);
                if (lUserRyt == 1)
                {
                    InfoLbl.Text = "Have User Right Info";
                }
                else if (lUserRyt == 0)
                {
                    InfoLbl.Text = "Don't have user right Info";
                }
            }
            
        }

        private void UserBtn_Click(object sender, EventArgs e)
        {
            ISecurity m_Security;
            m_Security = new SecurityClass();
            InfoLbl.Text = "";
            Securitylist.Items.Clear();
            Infolist.Items.Clear();
            string strUserName;
            string strUserSecurity="";
            strUserName = UsrNmeTxt.Text;
            if (strUserName == "")
            {
                MessageBox.Show("Please enter the user name");
            }
            else
            {
                m_Security.GetUserRightsInfo(strUserName, ref strUserSecurity);
                Securitylist.Items.Add(strUserSecurity);
            }
        }


    }
}